### PR TITLE
conftest: Qtile -> TestManager

### PR DESCRIPTION
**Pull request by m-col**
The Qtile class in test/conftest.py and the fixture 'qtile' that yields
it to tests share a name with the Qtile class in libqtile and the
'qtile' name that its instance takes. This has lead to confusion as they
do not represent the same thing, and are used in different ways. To make
the distinction more clear, this commit simply renames the testing
Qtile class to TestManager, and renames the fixture to 'm', so that
the manager passed to tests is referenced by the variable 'm'.

### CODE REVIEW

File: test/test_popup.py [hunk 1]
```diff
@@ -26,24 +26,24 @@
...
-    qtile.test_xeyes()
-    qtile.windows_map = {}
+@pytest.mark.parametrize("m", [BareConfig], indirect=True)
+def test_popup_focus(m):
+    m.test_xeyes()
+    m.windows_map = {}
 
-    # we have to add .conn so that Popup thinks this is libqtile.qtile
-    qtile.conn = xcbq.Connection(qtile.display)
+    # we have to add .conn so that Popup thinks this is libqtile.m
```
Comment (tcld) [inline test/test_popup.py:34]: Search and replace worked too well on this one. :)

**Comment by tcld** [comment]
Two small things:
  1. I myself prefer longer names for variables, especially if you ever plan to do a search (and replace). Thus I would choose something like `testmanager` instead of `m`.
  2. I would kind of like to see `conftest.py` renamed: We can't call it `test_manager.py` after its central class, but maybe `base.py`? Next to `test_config.py` it doesn't stand out as much as it should, I think.

File: test/test_manager.py [hunk 2]
```diff
@@ -87,111 +86,102 @@ class ManagerConfig(Config):
...
+    assert m.c.screen.info()["x"] == 0
+    assert m.c.screen.info()["width"] == 800
+    assert m.c.group.info()["name"] == 'a'
+    assert m.c.group.info()["focus"] == 'xclock'
 
 
 @pytest.mark.parametrize("xephyr", [{"xoffset": 0}], indirect=True)
 @manager_config
-def test_clone_dim(qtile):
-    self = qtile
```
Comment (tcld) [inline test/test_manager.py:123]: Thanks for removing **that**!

**Comment by ramnes** [comment]
`conftest.py` is a standard file used by pytest, you can't really change its name. I do agree on the longer name for the fixture, and I'd vouch for "manager".

**Comment by tcld** [comment]
I didn't know about `conftest.py`, I'm very new to pytest. Just forget about that part, then.

`manager` sounds ok to me. With a class whose purpose mostly is to communicate with a running Qtile instance, the name could even be `QtileClient` or something - as long as it isn't `Qtile` anymore. :)

**Comment by m-col** [comment]
Lol, nice spot.

File: test/test_popup.py (hunk 1 above)
Comment (m-col) [reply test/test_popup.py:34]: Lol yep

**Comment by m-col** [comment]
Alright how's that for you fellas

File: docs/manual/contributing.rst [hunk 3]
```diff
@@ -61,11 +61,11 @@ documentation, but there are tutorials online that explain how it is used.
 
 Our tests are written inside the ``test`` folder at the top level of the
 repository. Reading through these, you can get a feel for the approach we take
-to test a given unit. Most of the tests involve an object called ``qtile``
-(note that this is distinct from ``libqtile.qtile``).  This exposes a command
-client at ``qtile.c`` that we use to test a Qtile instance running in a
-separate thread as if we were using a command client from within a running
-Qtile session.
+to test a given unit. Most of the tests involve an object called ``m``. This
```
Comment (tcld) [inline docs/manual/contributing.rst:64]: `manager`, right?^^

Comment (m-col) [reply docs/manual/contributing.rst:64]: Ah yep, forgot to update this, thanks!

File: test/conftest.py [hunk 4]
```diff
@@ -495,23 +495,23 @@ def qtile(request, xephyr):
...
-            yield q
+            yield manager
         finally:
-            q.terminate()
+            manager.terminate()
 
 
 @pytest.fixture(scope="function")
-def qtile_nospawn(request, xephyr):
+def m_nospawn(request, xephyr):
```
Comment (tcld) [inline test/conftest.py:507]: Anoter `m` - honestly can't tell whether this one is inentional, though.

Comment (m-col) [reply test/conftest.py:507]: oh, no, that was a total miss by my using \<m\> as my search and replace. This means that this exists elsewhere too! Man, really not on the ball today. I will replace m_nospawn everywhere.

**Comment by tych0** [comment]
Thanks folks, I think this is a nice improvement.

**Pull request merged**